Missing update method on the CaseInsensitiveDict

We got mypy errors after the release 6 hours ago:

```
communication/slack/tasks.py:39: error: "CaseInsensitiveDict?[builtins.str]" has no attribute "update"
```

This is when we do (simplified):
```python
session = requests.Session()
session.headers.update({"Authorization": f"Bearer {access_token}"})
```

Also part of the docs: https://docs.python-requests.org/en/master/user/advanced/

I think the `CaseInsensitiveDict` should also contain the `.update()` method. Will open a PR in a second.

--- a/stdlib/imaplib.pyi
+++ b/stdlib/imaplib.pyi
@@ -2,7 +2,7 @@ import subprocess
 import sys
 import time
 from _typeshed import Self
-from builtins import list as _list # conflicts with a method named "list"
+from builtins import list as _list  # conflicts with a method named "list"
 from socket import socket as _socket
 from ssl import SSLContext, SSLSocket
 from types import TracebackType

--- a/stdlib/imaplib.pyi
+++ b/stdlib/imaplib.pyi
@@ -2,10 +2,11 @@ import subprocess
 import sys
 import time
 from _typeshed import Self
+from builtins import list as _list # conflicts with a method named "list"
 from socket import socket as _socket
 from ssl import SSLContext, SSLSocket
 from types import TracebackType
-from typing import IO, Any, Callable, Pattern, TypeVar
+from typing import IO, Any, Callable, Pattern
 from typing_extensions import Literal, TypeAlias
 
 __all__ = ["IMAP4", "IMAP4_stream", "Internaldate2tuple", "Int2AP", "ParseFlags", "Time2Internaldate", "IMAP4_SSL"]
@@ -15,9 +16,6 @@ __all__ = ["IMAP4", "IMAP4_stream", "Internaldate2tuple", "Int2AP", "ParseFlags"
 _CommandResults: TypeAlias = tuple[str, list[Any]]
 
 _AnyResponseData: TypeAlias = list[None] | list[bytes | tuple[bytes, bytes]]
-
-_T = TypeVar("_T")
-_list: TypeAlias = list[_T]  # conflicts with a method named "list"
 
 class IMAP4:
     error: type[Exception]

--- a/stubs/requests/requests/sessions.pyi
+++ b/stubs/requests/requests/sessions.pyi
@@ -1,14 +1,12 @@
 from _typeshed import Self, SupportsItems
-from typing import IO, Any, Callable, Iterable, Mapping, MutableMapping, TypeVar, Union
+from typing import IO, Any, Callable, Iterable, Mapping, MutableMapping, Union
 from typing_extensions import TypeAlias
 
-from urllib3 import _collections
+from urllib3._collections import _collections import RecentlyUsedContainer as RecentlyUsedContainer
 
-from . import adapters, auth as _auth, compat, cookies, exceptions, hooks, models, status_codes, structures, utils
+from . import adapters, auth as _auth, compat, cookies, exceptions, hooks, models, status_codes, utils
 from .models import Response
-
-_KT = TypeVar("_KT")
-_VT = TypeVar("_VT")
+from .structures import CaseInsensitiveDict as CaseInsensitiveDict
 
 _BaseAdapter = adapters.BaseAdapter
 OrderedDict = compat.OrderedDict
@@ -28,8 +26,6 @@ TooManyRedirects = exceptions.TooManyRedirects
 InvalidSchema = exceptions.InvalidSchema
 ChunkedEncodingError = exceptions.ChunkedEncodingError
 ContentDecodingError = exceptions.ContentDecodingError
-RecentlyUsedContainer: TypeAlias = _collections.RecentlyUsedContainer[_KT, _VT]
-CaseInsensitiveDict: TypeAlias = structures.CaseInsensitiveDict[_VT]
 HTTPAdapter = adapters.HTTPAdapter
 requote_uri = utils.requote_uri
 get_environ_proxies = utils.get_environ_proxies

--- a/stubs/requests/requests/sessions.pyi
+++ b/stubs/requests/requests/sessions.pyi
@@ -2,7 +2,7 @@ from _typeshed import Self, SupportsItems
 from typing import IO, Any, Callable, Iterable, Mapping, MutableMapping, Union
 from typing_extensions import TypeAlias
 
-from urllib3._collections import RecentlyUsedContainer as RecentlyUsedContainer
+from urllib3._collections import RecentlyUsedContainer
 
 from . import adapters, auth as _auth, compat, cookies, exceptions, hooks, models, status_codes, utils
 from .models import Response

--- a/stubs/requests/requests/sessions.pyi
+++ b/stubs/requests/requests/sessions.pyi
@@ -2,7 +2,7 @@ from _typeshed import Self, SupportsItems
 from typing import IO, Any, Callable, Iterable, Mapping, MutableMapping, Union
 from typing_extensions import TypeAlias
 
-from urllib3._collections import _collections import RecentlyUsedContainer as RecentlyUsedContainer
+from urllib3._collections import RecentlyUsedContainer as RecentlyUsedContainer, _collections, import
 
 from . import adapters, auth as _auth, compat, cookies, exceptions, hooks, models, status_codes, utils
 from .models import Response

--- a/stubs/requests/requests/sessions.pyi
+++ b/stubs/requests/requests/sessions.pyi
@@ -2,7 +2,7 @@ from _typeshed import Self, SupportsItems
 from typing import IO, Any, Callable, Iterable, Mapping, MutableMapping, Union
 from typing_extensions import TypeAlias
 
-from urllib3._collections import RecentlyUsedContainer as RecentlyUsedContainer, _collections, import
+from urllib3._collections import RecentlyUsedContainer as RecentlyUsedContainer
 
 from . import adapters, auth as _auth, compat, cookies, exceptions, hooks, models, status_codes, utils
 from .models import Response

